File: negocio-tundama-backend/src/main/java/com/tundama/negociotundamabackend/model/entity/Usuario.java
package com.tundama.negociotundamabackend.model.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.time.LocalDateTime;

@Entity
@Table(name = "usuarios")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Usuario {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "id_usuario", unique = true, nullable = false, length = 50)
    private String idUsuario;  // Formato: NTU-2023-ABR-0074

    @Column(name = "nombre", nullable = false, length = 100)
    private String nombre;

    @Column(name = "direccion", length = 200)
    private String direccion;

    @Column(name = "email", unique = true, nullable = false, length = 100)
    private String email;

    @Column(name = "contrasena", nullable = false)
    private String contrasena;

    @Column(name = "notas", length = 500)
    private String notas;

    @Column(name = "fecha_registro")
    private LocalDateTime fechaRegistro;

    @Column(name = "activo")
    private Boolean activo = true;

    @PrePersist
    public void prePersist() {
        fechaRegistro = LocalDateTime.now();
    }
}

File: Version Android/lib/models/usuario.dart
// ==================== lib/models/usuario.dart ====================
class Usuario {
  final int? id;
  final String idUsuario;
  final String nombre;
  final String? direccion;
  final String email;
  final String? contrasena;
  final String? notas;
  final DateTime? fechaRegistro;
  final bool activo;

  Usuario({
    this.id,
    required this.idUsuario,
    required this.nombre,
    this.direccion,
    required this.email,
    this.contrasena,
    this.notas,
    this.fechaRegistro,
    this.activo = true,
  });

  factory Usuario.fromJson(Map<String, dynamic> json) {
    return Usuario(
      id: json['id'],
      idUsuario: json['idUsuario'] ?? '',
      nombre: json['nombre'] ?? '',
      direccion: json['direccion'],
      email: json['email'] ?? '',
      notas: json['notas'],
      fechaRegistro: json['fechaRegistro'] != null 
          ? DateTime.parse(json['fechaRegistro']) 
          : null,
      activo: json['activo'] ?? true,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'idUsuario': idUsuario,
      'nombre': nombre,
      'direccion': direccion,
      'email': email,
      'contrasena': contrasena,
      'notas': notas,
    };
  }
}
